sample/src/main/java/org/smartregister/immunization/sample/MainActivity.java

```diff
@@ -531,6 +532,119 @@ public void startServices() {
+                return vaccineGroup;
+            }
+        }
+
+        return null;
+    }
+
+    private void updateVaccineGroupsUsingAlerts(List<String> affectedVaccines, List<Vaccine> vaccineList, List<Alert> alerts) {
+        if (affectedVaccines != null && vaccineList != null) {
+            // Update all other affected vaccine groups
+            HashMap<VaccineGroup, ArrayList<VaccineWrapper>> affectedGroups = new HashMap<>();
+            for (String curAffectedVaccineName : affectedVaccines) {
+                boolean viewFound = false;
+                // Check what group it is in
+                for (VaccineGroup curGroup : vaccineGroups) {
+                    ArrayList<VaccineWrapper> groupWrappers = curGroup.getAllVaccineWrappers();
+                    if (groupWrappers == null) groupWrappers = new ArrayList<>();
+                    for (VaccineWrapper curWrapper : groupWrappers) {
+                        String curWrapperName = curWrapper.getName();
+
+                        // Check if current wrapper is one of the combined vaccines
+                        if (ImmunizationLibrary.getInstance().COMBINED_VACCINES.contains(curWrapperName)) {
+                            // Check if any of the sister vaccines is currAffectedVaccineName
+                            String[] allSisters = ImmunizationLibrary.getInstance().COMBINED_VACCINES_MAP.get(curWrapperName).split(" / ");
+                            for (int i = 0; i < allSisters.length; i++) {
+                                if (allSisters[i].replace(" ", "").equalsIgnoreCase(curAffectedVaccineName.replace(" ", ""))) {
+                                    curWrapperName = allSisters[i];
+                                    break;
+                                }
+                            }
+                        }
+
+                        if (curWrapperName.replace(" ", "").toLowerCase()
+                                .contains(curAffectedVaccineName.replace(" ", "").toLowerCase())) {
+                            if (!affectedGroups.containsKey(curGroup)) {
+                                affectedGroups.put(curGroup, new ArrayList<VaccineWrapper>());
+                            }
+
+                            affectedGroups.get(curGroup).add(curWrapper);
+                            viewFound = true;
+                        }
+
+                        if (viewFound) break;
+                    }
+
+                    if (viewFound) break;
+                }
+            }
+
+            for (VaccineGroup curGroup : affectedGroups.keySet()) {
+                try {
+                    vaccineGroups.remove(curGroup);
+                    addVaccineGroup(Integer.valueOf((String) curGroup.getTag(R.id.vaccine_group_parent_id)),
+                            //TODO if error use immediately below
+                            // (org.smartregister.immunization.domain.jsonmapping.VaccineGroup) curGroup.getTag(R.id.vaccine_group_vaccine_data),
```

Comment:
Will this `TODO` be handled on this PR?

Reply:
This is already part of Master and was not introduced by this PR . See [MainActivity.java#L841](https://github.com/opensrp/opensrp-client-immunization/blob/5478d3a34ecb33b9a285d122e57594d9dc475127/sample/src/main/java/org/smartregister/immunization/sample/MainActivity.java#L841)